ldap-api/src/main/java/ca/bc/gov/hlth/ldapapi/service/UserService.java

```diff
@@ -0,0 +1,148 @@
+package ca.bc.gov.hlth.ldapapi.service;
+
+import ca.bc.gov.hlth.ldapapi.model.UserCredentials;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.SearchResult;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class UserService {
+
+    private final LdapService ldapService;
+
+    private OrgLookup orgLookup;
+
+    @Value("${ldap.attempt.timeout}")
+    int attemptTimeout;
+
+    @Value("${roleAttributeNames}")
+    private String roles;
+
+    final ConcurrentHashMap<String, LoginAttempts> loginAttemptsMap = new ConcurrentHashMap<>();
+
+    public UserService(LdapService ldapService, OrgLookup orgLookup) {
+        this.ldapService = ldapService;
+        this.orgLookup = orgLookup;
+    }
+
+    public Object authenticate(UserCredentials userCredentials) throws NamingException {
+        SearchResult userInfo = ldapService.searchUser(userCredentials.getUserName());
+        if (userInfo == null) {
+            return null;
+        }
+
+        boolean userUnlocked = ldapService.checkUserLocked(userInfo.getAttributes());
+
+        boolean validCredentials = false;
+        long lockoutTimeInHours = 0;
+        int remainingAttempts = 3;
+
+        if (userUnlocked) {
+            validCredentials = ldapService.authenticateUser(userInfo.getName(), userCredentials.getPassword());
+
+            if (!validCredentials) {
+                userUnlocked = !updateUserFailedLoginAttempts(userInfo.getName());
```

Comment:
With all the previous `AuthenticationResult` checks gone, this can just move down into the second `if (!validCredentials) {...}` below